Does Not Support All Unicode Chars

**Describe the bug**
When putting a URL into the "Query sample input" text field (the address bar of the app), if that URL has Unicode characters in it, the whole app crashes and removes itself from the DOM and goes to a pseudo white screen.

**To Reproduce**
Steps to reproduce the behavior:
1. Position your cursor in the address bar of the app
2. Put ANY emoji in it, in any position

**Expected behavior**
The emoji stays in the URL bar and can be queried against since Entra ID supports full Unicode chars in their properties.
The app doesn't crash ultimately.

**Screenshots**
(sensitive data is redacted by photo edits)
Before Emoji in the address bar is injected:
![Screenshot of the query bar showing a filtered group with an emoji prompt pulled up but no emoji.](https://github.com/microsoftgraph/microsoft-graph-explorer-v4/assets/5030490/56d6c680-3f25-4bde-b032-106304e3ca06)

After:
![Screenshot of the App's crashed state](https://github.com/microsoftgraph/microsoft-graph-explorer-v4/assets/5030490/662a0705-bbe9-495f-a752-d11e127dc2a1)

**Desktop (please complete the following information):**
 - OS: Windows 11, 22H2
 - Browser Microsoft Edge
 - Version 116.0.1938.62

**Additional context**
console logs:
``` JavaScript
react-dom.production.min.js:4014  Error: parser.js: initializeInputChars(): input length out of range: 180
    at N (apg-api-bundle.js:5785:115)
    at t.exports.parse (apg-api-bundle.js:5916:21)
    at e.value (abnf.ts:43:40)
    at auto-complete.util.ts:62:32
    at Pf (auto-complete.util.ts:50:17)
    at Zf (AutoComplete.tsx:290:24)
    at ka (react-dom.production.min.js:3436:7)
    at Ts (react-dom.production.min.js:4165:7)
    at ku (react-dom.production.min.js:6626:86)
    at bl (react-dom.production.min.js:6283:11)
hs @ react-dom.production.min.js:4014
n.callback @ react-dom.production.min.js:4031
Uo @ react-dom.production.min.js:2972
_u @ react-dom.production.min.js:5670
bu @ react-dom.production.min.js:5635
Au @ react-dom.production.min.js:5610
(anonymous) @ react-dom.production.min.js:6365
Cl @ react-dom.production.min.js:6324
ll @ react-dom.production.min.js:6039
zi @ react-dom.production.min.js:2538
dl @ react-dom.production.min.js:6061
Le @ react-dom.production.min.js:736
Wr @ react-dom.production.min.js:2087
qt @ react-dom.production.min.js:1285
jt @ react-dom.production.min.js:1267
apg-api-bundle.js:5785  Uncaught Error: parser.js: initializeInputChars(): input length out of range: 180
    at N (apg-api-bundle.js:5785:115)
    at t.exports.parse (apg-api-bundle.js:5916:21)
    at e.value (abnf.ts:43:40)
    at auto-complete.util.ts:62:32
    at Pf (auto-complete.util.ts:50:17)
    at Zf (AutoComplete.tsx:290:24)
    at ka (react-dom.production.min.js:3436:7)
    at Ts (react-dom.production.min.js:4165:7)
    at ku (react-dom.production.min.js:6626:86)
    at bl (react-dom.production.min.js:6283:11)
N @ apg-api-bundle.js:5785
parse @ apg-api-bundle.js:5916
value @ abnf.ts:43
(anonymous) @ auto-complete.util.ts:62
Pf @ auto-complete.util.ts:50
Zf @ AutoComplete.tsx:290
ka @ react-dom.production.min.js:3436
Ts @ react-dom.production.min.js:4165
ku @ react-dom.production.min.js:6626
bl @ react-dom.production.min.js:6283
yl @ react-dom.production.min.js:6274
vl @ react-dom.production.min.js:6258
al @ react-dom.production.min.js:5978
ll @ react-dom.production.min.js:6033
zi @ react-dom.production.min.js:2538
dl @ react-dom.production.min.js:6061
Le @ react-dom.production.min.js:736
Wr @ react-dom.production.min.js:2087
qt @ react-dom.production.min.js:1285
jt @ react-dom.production.min.js:1267
defaultWorkerFactory.js:23  Refused to create a worker from 'https://developer.microsoft.com/json.worker.js' because it violates the following Content Security Policy directive: "worker-src data:".

(anonymous) @ defaultWorkerFactory.js:23
e @ defaultWorkerFactory.js:66
value @ defaultWorkerFactory.js:124
n @ simpleWorker.js:238
value @ editorWorkerServiceImpl.js:545
value @ editorWorkerServiceImpl.js:557
value @ webWorker.js:50
value @ webWorker.js:85
e._getClient @ workerManager.js:55
e.getLanguageServiceWorker @ workerManager.js:66
i @ jsonMode.js:19
e._resetSchema @ languageFeatures.js:73
(anonymous) @ languageFeatures.js:41
value @ event.js:566
value @ modelServiceImpl.js:445
(anonymous) @ modelServiceImpl.js:308
(anonymous) @ modelServiceImpl.js:71
value @ event.js:566
value @ textModel.js:266
(anonymous) @ index.mjs:373
tu @ react-dom.production.min.js:5051
ru @ react-dom.production.min.js:5176
wl @ react-dom.production.min.js:6418
(anonymous) @ react-dom.production.min.js:6347
C @ scheduler.production.min.js:94
I @ scheduler.production.min.js:125
simpleWorker.js:27  Failed to construct 'Worker': Access to the script at 'https://developer.microsoft.com/json.worker.js' is denied by the document's Content Security Policy.
He @ simpleWorker.js:27
value @ editorWorkerServiceImpl.js:547
value @ editorWorkerServiceImpl.js:557
value @ webWorker.js:50
value @ webWorker.js:85
e._getClient @ workerManager.js:55
e.getLanguageServiceWorker @ workerManager.js:66
i @ jsonMode.js:19
e._resetSchema @ languageFeatures.js:73
(anonymous) @ languageFeatures.js:41
value @ event.js:566
value @ modelServiceImpl.js:445
(anonymous) @ modelServiceImpl.js:308
(anonymous) @ modelServiceImpl.js:71
value @ event.js:566
value @ textModel.js:266
(anonymous) @ index.mjs:373
tu @ react-dom.production.min.js:5051
ru @ react-dom.production.min.js:5176
wl @ react-dom.production.min.js:6418
(anonymous) @ react-dom.production.min.js:6347
C @ scheduler.production.min.js:94
I @ scheduler.production.min.js:125
editorSimpleWorker.js:810  Uncaught (in promise) Error: Unexpected usage
    at e.value (editorSimpleWorker.js:810:29)
    at webWorker.js:52:24
```

--- a/src/app/views/query-runner/query-input/auto-complete/auto-complete.util.ts
+++ b/src/app/views/query-runner/query-input/auto-complete/auto-complete.util.ts
@@ -61,11 +61,8 @@ function getValidationError(queryUrl: string): string | null {
   try {
     const validator = new ValidatedUrl();
     const validation = validator.validate(queryUrl);
-    if (!validation.success) {
-      return queryUrl.substring(validation.matched, validation.maxMatched);
-    } else {
-      return null;
-    }
+    return (!validation.success) ?
+      queryUrl.substring(validation.matched, validation.maxMatched) : null;
   } catch (error) {
     return null;
   }

--- a/src/app/views/query-runner/query-input/auto-complete/auto-complete.util.ts
+++ b/src/app/views/query-runner/query-input/auto-complete/auto-complete.util.ts
@@ -58,12 +58,16 @@ function getErrorMessage(queryUrl: string) {
 }
 
 function getValidationError(queryUrl: string): string | null {
-  const validator = new ValidatedUrl();
-  const validation = validator.validate(queryUrl);
-  if (!validation.success) {
-    return queryUrl.substring(validation.matched, validation.maxMatched);
+  try {
+    const validator = new ValidatedUrl();
+    const validation = validator.validate(queryUrl);
+    if (!validation.success) {
+      return queryUrl.substring(validation.matched, validation.maxMatched);
+    }
+    throw new Error('Validation unsuccessful');
+  } catch (error) {
+    return null;
   }
-  return null;
 }
 
 function getSearchText(input: string, index: number) {

--- a/src/app/views/query-runner/query-input/auto-complete/auto-complete.util.ts
+++ b/src/app/views/query-runner/query-input/auto-complete/auto-complete.util.ts
@@ -51,7 +51,7 @@ function getErrorMessage(queryUrl: string) {
   if (error) {
     return `${translateMessage('Possible error found in URL near')}: ${error}`;
   }
-  if (queryUrl.indexOf('graph.microsoft.com') === -1){
+  if (queryUrl.indexOf('graph.microsoft.com') === -1) {
     return translateMessage('The URL must contain graph.microsoft.com');
   }
   return '';
@@ -63,8 +63,9 @@ function getValidationError(queryUrl: string): string | null {
     const validation = validator.validate(queryUrl);
     if (!validation.success) {
       return queryUrl.substring(validation.matched, validation.maxMatched);
+    } else {
+      return null;
     }
-    throw new Error('Validation unsuccessful');
   } catch (error) {
     return null;
   }

